src/common/services/melon/resolve.ts

```diff
@@ -13,15 +12,15 @@ const getTitle = (document_: Document) => {
 const getArtists = (document_: Document) => {
   return [...document_.querySelectorAll('.section_info .info .artist a')]
     .map((element) => element.textContent?.trim())
-    .filter(isDefined)
+    .filter((a) => a != undefined)
```

Comment:
```suggestion
    .filter(isDefined)
```

here's a case where we need isDefined

since TS isn't smart enough yet to realize that `a != undefined` is filtering out all undefined elements, it thinks the return type of this function is `(string | undefined)[]`

if we switch it out for `isDefined` which is a type guard, it correctly sets the return type as `string[]` and removes the need to cast the return of the `resolve` function to `ResolveData` (line 86)